File: nave-frontEnd/src/app/modelo/notripulada.ts
export class Notripulada {

    _id: string;
    toneladasEmpuje: string;
    velocidad: string;
    nave: string;
   
    
    constructor(toneladasEmpuje: string, velocidad: string, _id: string,  nave: string ){
        this.toneladasEmpuje = toneladasEmpuje;
        this.velocidad = velocidad;
        this._id = _id;
        this.nave = nave;    
    }
}

File: nave-backend/src/main/java/reto/sofka2022/navebackend/modelos/NoTripulada.java
package reto.sofka2022.navebackend.modelos;

//Componentes externos
import lombok.Data;
import lombok.Setter;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AccessLevel;

@Document(collection = "tb-NoTripulada")
@Data

public class NoTripulada {
    @Id
    @Setter(AccessLevel.NONE) //le establezco acces none para que no pueda modificarse el id
    
    private String _id;
    private int toneladasEmpuje;
    private int velocidad;
    @DBRef private Nave nave;
}
